factory.go

```diff
@@ -28,21 +38,67 @@ func (f factory) DeploymentType() deployer.DeploymentType {
 	return "python"
 }
 
-func (f factory) ConfigurationSchema() *schema.TypedScopeSchema[*Config] {
+func (f factory) ConfigurationSchema() *schema.TypedScopeSchema[*config.Config] {
 	return Schema
 }
 
-func (f factory) Create(config *Config, logger log.Logger) (deployer.Connector, error) {
+func (f factory) NextConnectorIndex() int {
+	return int(atomic.AddInt64(f.connectorCounter, 1))
+}
+
+func (f factory) Create(config *config.Config, logger log.Logger) (deployer.Connector, error) {
 	pythonPath, err := binaryCheck(config.PythonPath)
 	if err != nil {
-		return &Connector{}, fmt.Errorf("python binary check failed with error: %w", err)
+		return &connector.Connector{}, fmt.Errorf("python binary check failed with error: %w", err)
+	}
+
+	pythonSemver := config.PythonSemVer
+	if pythonSemver == "" {
+		outputSemver, err := f.parsePythonVersion(pythonPath)
+		if err != nil {
+			return nil, err
+		}
+		pythonSemver = *outputSemver
+	}
+
+	connectorFilename := strings.Join([]string{
+		"connector",
+		strings.Replace(pythonSemver, ".", "-", -1),
+		strconv.Itoa(f.NextConnectorIndex())},
+		"_")
+
+	absWorkDir, err := filepath.Abs(config.WorkDir)
+	if err != nil {
+		return nil, fmt.Errorf("error determining absolute path for python deployer's working directory (%w)", err)
```

Comment:
If `Abs` fails, the doc doesn't appear to say exactly what `err` will be: does it have enough information, or should we report `config.WorkDir` here?